{"cert_user_id": "efranelas@github", "next_comment_id": 2, "comment": {"1604067689_mirrored_Noruka_github": [{"comment_id": 1, "body": "My /opt/zeronet/src/Config.py has no getCmdlineValue method and my zeronet version is [0.7.1-1](https://aur.archlinux.org/packages/zeronet/)\r\n```\r\nself.version = \"0.7.0\"\r\nself.rev = 4206\r\n```\r\n\r\nSeems not like an OS problem, but I am using the latest arch linux\r\n\r\nMy `journalctl -xe` output\r\n```\r\nJan 14 17:24:04 archer zeronet[13806]: [17:24:04] - Libsecpk256k1 loaded: CompiledLib in 1.170s\r\nJan 14 17:24:05 archer zeronet[13806]: [17:24:05] - Unhandled exception: 'ConfigPlugin' object has no attribute 'getCmdlineValue'\r\nJan 14 17:24:05 archer zeronet[13806]: Traceback (most recent call last):\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"zeronet.py\", line 16, in main\r\nJan 14 17:24:05 archer zeronet[13806]:     import main\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/src/main.py\", line 64, in <module>\r\nJan 14 17:24:05 archer zeronet[13806]:     config.loadPlugins()\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/src/Config.py\", line 464, in loadPlugins\r\nJan 14 17:24:05 archer zeronet[13806]:     ConfigPlugin(self)\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/src/Config.py\", line 459, in __init__\r\nJan 14 17:24:05 archer zeronet[13806]:     self.createArguments()\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/plugins/Zeroname/UiRequestPlugin.py\", line 30, in createArguments\r\nJan 14 17:24:05 archer zeronet[13806]:     return super(ConfigPlugin, self).createArguments()\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/plugins/OptionalManager/OptionalManagerPlugin.py\", line 239, in createArguments\r\nJan 14 17:24:05 archer zeronet[13806]:     return super(ConfigPlugin, self).createArguments()\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/plugins/Bigfile/BigfilePlugin.py\", line 791, in createArguments\r\nJan 14 17:24:05 archer zeronet[13806]:     return super(ConfigPlugin, self).createArguments()\r\nJan 14 17:24:05 archer zeronet[13806]:   File \"/opt/zeronet/plugins/Benchmark/BenchmarkPlugin.py\", line 347, in createArguments\r\nJan 14 17:24:05 archer zeronet[13806]:     if self.getCmdlineValue(\"test\") == \"benchmark\":\r\nJan 14 17:24:05 archer zeronet[13806]: AttributeError: 'ConfigPlugin' object has no attribute 'getCmdlineValue'\r\nJan 14 17:24:05 archer zeronet[13806]: - Starting ZeroNet...\r\nJan 14 17:24:05 archer zeronet[13806]: ---\r\nJan 14 17:24:05 archer zeronet[13806]: Please report it: https://github.com/HelloZeroNet/ZeroNet/issues/new?assignees=&labels=&template=bug-repo>\r\nJan 14 17:24:05 archer systemd[1]: zeronet.service: Succeeded.\r\n```\r\n\r\nEDIT: Even after removing /opt/zeronet and reinstalling, the Config.py file is still the same with no getCmdlineValue method. However I've found another Config.py in my data_dir  `/var/lib/zeronet/1uPDaT3uSyWAPdCv1WkMb5hBQjWSNNACf/core/src/Config.py` that:\r\n- has the getCmdlineValue method in line 465\r\n- self.version = \"0.7.1\"\r\n- self.rev = 4325\r\n\r\nI don't know if it is safe to move the second Config.py to /opt/zeronet/src/Config.py but I just did it and the log shows a successful start.", "added": 1579023552, "modified": 1579026231, "source_link": "https://github.com/HelloZeroNet/ZeroNet/issues/2336#issuecomment-574363107", "source_type": "github"}]}}